### PR TITLE
Setting up new project with mailhog without docker breaks use of mailhog 

### PR DESCRIPTION
**Pull request by shireenrao**
The recent addition of mailhog as a docker container broke the template if one wanted to use mailhog without docker locally. As EMAIL_BACKEND was set, mailhog was not getting the emails from the django app. I fixed it by adding special cases in {{cookiecutter.project_slug}}/config/settings/local.py. I have now tested all 4 combinations of mailhog (y or n) and docker (y or n) and can confirm that they all work. As mailhog now works with docker, I also removed the warning message in hooks/post_gen_project.py which said mailhog and docker were not compatible.
The last change I made was a cosmetic one in dev.yml to remove unnecessary whitespace.

### CODE REVIEW

**Comment by jayfk** [comment]
Thanks for this pull request!

Could you please add the commits in #555 and #554 to this PR? Having them shattered across multiple PRs makes this hard to follow.

**Comment by shireenrao** [comment]
Apologies for not doing it in one PR. I think I added the commits in #555 and #554 in this PR correctly, but in case I did not, please let me know.

**Comment by pydanny** [comment]
@jayfk Is this ready to merge?

**Comment by jayfk** [comment]
I'm sorry, I haven't had the time to review it :/

**Comment by jayfk** [comment]
🔔 @noisy 

Got some time to take a look? This builds on top of your work.

**Comment by jayfk** [comment]
Small update: I couldn't make it work. I'm a bit in a rush right now, I'll check again later.

Sorry for keeping this open for so long.

**Comment by shireenrao** [comment]
Hi @jayfk - let me know if I can help.

**Comment by pydanny** [comment]
:bell: Any updates? Is this ready to go?

**Comment by ssteinerx** [comment]
Coming up on three months.  Has conflicts.  

@pydanny Is this something you want to merge?  Should we ask author to resolve conflicts or just close?

**Comment by shireenrao** [comment]
I can take a look at the conflicts..

**Comment by shireenrao** [comment]
take a look now @ssteinerx @pydanny

**Comment by shireenrao** [comment]
I just saw that I should remove the changes I made to the file docs/developing-locally-docker.rst as it has further been changed since I made my changes. It's done and submitted.

**Comment by webyneter** [comment]
@shireenrao, what's the status on this?

**Comment by shireenrao** [comment]
@webyneter not sure what's going on with this.. last I checked a core developer needs to test the changes. It's been a while.. so a lot may have changed since. I can check and get back if it is still an issue.

**Comment by webyneter** [comment]
@shireenrao, that would be nice :)

**Comment by webyneter** [comment]
Confirmed to work on my host. Merging.
